### PR TITLE
Auto attack

### PR DESCRIPTION
**Pull request by B0ZIA**
Clicking in screen for auto-attack System

### CODE REVIEW

File: Scripts/StaminaSystem.gd [hunk 1]
```diff
@@ -0,0 +1,43 @@
+extends Label
+
+onready var stamina_label: Label = get_node(".")
```
Comment (Ajver) [inline Scripts/StaminaSystem.gd]:
```gdscript
get_node(".")
```
Is the same as:
```gdscript
self
```

File: Scripts/StaminaSystem.gd [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+var auto_atack: bool = true
+var timer = null
+
+func _ready():
+	max_stamina = ElfStats.get_stat("stamina").value
+	current_stamina = max_stamina
+	set_process_input(true)
+
+func _input(event):
+	if event is InputEventMouseButton and event.pressed and event.button_index == 1:
```
Comment (Ajver) [inline Scripts/StaminaSystem.gd]: I think, it's better to set custom action in project settings and use Input.is_action_just_pressed("our_action_name")

File: Scripts/StaminaSystem.gd [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		increasing_stamina()
+
+func increasing_stamina():
```
Comment (Ajver) [inline Scripts/StaminaSystem.gd]: So many code. Isn't that better?
```gdscript
current_stamina = min(current_stamina + 1, max_stamina)
```

anyway, 'else' statement in 26 line is useless. Why we setting current stamina to the max value, if it equals that value?

File: Scripts/StaminaSystem.gd [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		current_stamina = max_stamina
+	
+
+func _process(delta):
+	if current_stamina > 0:
+		auto_atack = true
+		if current_stamina - delta < 0:
+			current_stamina = 0
+		else:
+			current_stamina -= delta
```
Comment (Ajver) [inline Scripts/StaminaSystem.gd]: Maybe:
```gdscript
current_stamina = max(current_stamina - delta, 0)
```

File: Scripts/StaminaSystem.gd [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	else:
+		auto_atack = false
+	
+	max_stamina = ElfStats.get_stat("stamina").value
+	var status = String(stepify(current_stamina,0.1))+"s / " + String(max_stamina) + "s"
```
Comment (Ajver) [inline Scripts/StaminaSystem.gd]: `max_stamina` may be in stepify function as well. Don't You think?

File: Scripts/StaminaSystem.gd (hunk 3 above)
Comment (B0ZIA) [reply Scripts/StaminaSystem.gd]: You are right

File: Scripts/StaminaSystem.gd (hunk 5 above)
Comment (B0ZIA) [reply Scripts/StaminaSystem.gd]: W will add it

**Pull request merged**